app/controllers/dashboard/locations_controller.rb

```diff
@@ -79,10 +80,14 @@ def set_opening_times
     end
 
     def location_params
-      params.require(:location)
-            .permit(:name, :description, :address, :latitude, :longitude,
-                    opening_times_attributes: \
-                      %i[id location_id day opens_at closes_at closed open_24h])
+      current_params = params.require(:location)
+                             .permit(:name, :description, :address, :latitude, :longitude,
+                                     opening_times_attributes: \
+                                     %i[id location_id day opens_at closes_at closed open_24h])
+      return current_params unless params[:no_opening_times] == "yes"
+
+      # remove opening times if we don't need them
+      current_params.except(:opening_times_attributes)
```

Comment:
I find this slightly hard to read. What something like this work better?

```
def location_params
  attributes = %i[name description address latitude longitude]

  unless params[:no_opening_times] == "yes"
    attributes.push(opening_times_attributes: %i[id location_id day opens_at closes_at closed open_24h])
  end

  params.require(:location).permit(attributes)
end
```

Reply:
Thanks for sharing the article, I didn't know about the `_destroy` attribute. Added that now - it makes it much cleaner and also prevents some potential nasty bugs that could have occurred when updating an existing location and removing the opening times (since the update and delete operations were not atomic).

Also simplifies `def location_params` :-)

Updated in [83fddc7](https://github.com/mapzy/mapzy/pull/64/commits/83fddc74ce9709f8ec1cec0836f2191e3f0f0b18)